### PR TITLE
Enable Go driver's integration tests

### PR DESCRIPTION
**Pull request by noisersup**
Refs #144.

### CODE REVIEW

**Comment by noisersup** [comment]
I did checkout on #161 and used regex to cover most of the test cases.

The problem now is that one of the tests panic because of the time out.
```
go.mongodb.org/mongo-driver/mongo/integration/TestClient/RTT90_used_to_prevent_sending_requests unknown:
        === RUN   TestClient/RTT90_used_to_prevent_sending_requests
        panic: test timed out after 2m0s

        goroutine 1383386 [running]:
        testing.(*M).startAlarm.func1()
                /nix/store/9qcr8mh3c9vvn2x20d2ngqryhmvpcy49-go-1.18rc1/share/go/src/testing/testing.go:2029 +0x8e
        created by time.goFunc
                /nix/store/9qcr8mh3c9vvn2x20d2ngqryhmvpcy49-go-1.18rc1/share/go/src/time/sleep.go:176 +0x32
```

File: tests/mongo-go-driver.yml [hunk 1]
```diff
@@ -119,3 +212,372 @@ results:
...
+      - go.mongodb.org/mongo-driver/mongo/integration/unified/TestUnifiedSpec/unified-test-format/valid-pass/poc-retryable-reads.json/Aggregate_succeeds_after_InterruptedAtShutdown
+      - go.mongodb.org/mongo-driver/mongo/integration/unified/TestUnifiedSpec/unified-test-format/valid-pass/poc-retryable-reads.json/Find_fails_on_first_attempt
+      - go.mongodb.org/mongo-driver/mongo/integration/unified/TestUnifiedSpec/unified-test-format/valid-pass/poc-retryable-reads.json/Find_succeeds_on_second_attempt
+      - go.mongodb.org/mongo-driver/mongo/integration/unified/TestUnifiedSpec/unified-test-format/valid-pass/poc-retryable-reads.json/ListDatabases_succeeds_on_second_attempt
+      - go.mongodb.org/mongo-driver/mongo/integration/unified/TestUnifiedSpec/unified-test-format/valid-pass/poc-sessions.json
+      - go.mongodb.org/mongo-driver/mongo/integration/unified/TestUnifiedSpec/unified-test-format/valid-pass/poc-sessions.json/Server_supports_explicit_sessions
+      - go.mongodb.org/mongo-driver/mongo/integration/unified/TestUnifiedSpec/unified-test-format/valid-pass/poc-sessions.json/Server_supports_implicit_sessions
```
Comment (noisersup) [inline tests/mongo-go-driver.yml:580]: There's still a lot of mess in the pass section. I'm not sure if I'm doing everything correctly, so I will be happy if anyone take a look and tell me if there is any implemented way to reduce amount of lines here or should I make another PR which will allow us to match testnames with regex?

For now I tried to pass "/" at the end of the testname to match all tests under it but it didn't work as I expected.

**Comment by noisersup** [comment]
I merged #170.

I noticed that dance [shows some tests as unexpected](https://github.com/FerretDB/dance/blob/e67fee2bd2bc56046fb6ba53e74f260459e74269/out2.log#L42-L45) despite the output_regex: `.*skipping due to environmental constraints:.*`.

(I will remove *.log files before merge)

**Comment by noisersup** [comment]
Debugged it and it's probably because of [this pointer](https://github.com/FerretDB/dance/pull/153/files#diff-f93d5265aced90cffa2af511f5b4994b77f48d8f73d607e04da8f2db1dfecf88R196) :)

**Comment by noisersup** [comment]
[Example of mutable test](https://github.com/FerretDB/dance/actions/runs/2704130010/attempts/1)

**Comment by noisersup** [comment]
oh right

**Comment by AlekSi** [comment]
@noisersup Actually, don't do that, it is already in progress :) See #176

**Pull request merged**